tests/test_renderer.py

```diff
@@ -403,3 +406,41 @@ def test_render_innergroup(app):
                for el in render(app, compound_def,
                                 compound_parser=mock_compound_parser,
                                 options=['inner']))
+
+def test_resolve_function(app):
+    from breathe.directives import DoxygenFunctionDirective
+    from breathe.project import ProjectInfoFactory
+    from breathe.parser import DoxygenParserFactory
+    from breathe.finder.factory import FinderFactory
+    from docutils.statemachine import StringList
+    
+    app.config.breathe_separate_member_pages = False
+    app.config.breathe_default_project = 'test_project'
+    app.config.breathe_domain_by_extension = {}
+    app.config.breathe_domain_by_file_pattern = {}
+    app.config.breathe_use_project_refids = False
+    project_info_factory = ProjectInfoFactory(app)
+    parser_factory = DoxygenParserFactory(app)
+    finder_factory = FinderFactory(app, parser_factory)
+    cls_args = ('doxygenclass', ['at::Tensor'], 
+                {'members': '', 'protected-members': None, 'undoc-members': None},
+                StringList([], items=[]),
+                20, 24,
+                ('.. doxygenclass:: at::Tensor\n   :members:\n'
+                        '   :protected-members:\n   :undoc-members:'),
+                MockState(app),
+                MockStateMachine(),
+               ) 
+    cls = DoxygenFunctionDirective(finder_factory, project_info_factory, parser_factory, *cls_args)
+    with open(os.path.join(os.path.dirname(__file__), 'data', 'memberdef.xml')) as fid:
+        xml = fid.read()
+    doc = minidom.parseString(xml)
```

Comment:
I don't mind doing this here, but it kinda changes this file from a renderer only test into something that also relies on the parser performing in a correct way...

Reply:
what is the alternative?

Reply:
> what is the alternative?

Mocking.